servers/visual/visual_server_scene.cpp

```diff
@@ -608,7 +608,8 @@ void VisualServerScene::instance_set_blend_shape_weight(RID p_instance, int p_sh
 	if (instance->update_item.in_list()) {
 		_update_dirty_instance(instance);
 	}
-
+	//let it grow as big as possible, we don't want to lose values on resize
+	instance->blend_values.resize(MAX(instance->blend_values.size(), p_shape + 1));
```

Comment:
Probably also good to put in its own PR/commit.

Reply:
this should tehcnically not be needed, because when you load the MeshInstance, these are loaded after the mesh is set (_update_dirty_instance above will take care of it), so there should not be a problem. I think you probably tried to work around a bug I fixed close to the 3.1 release, so I would dare say this is no longer needed.